File: Graph/ScriptTemplate/GreaterThan/CGGreaterThan.lua
local CGGreaterThan = CGGreaterThan or {}
CGGreaterThan.__index = CGGreaterThan

function CGGreaterThan.new()
    local self = setmetatable({}, CGGreaterThan)
    self.inputs = {}
    return self
end

function CGGreaterThan:setInput(index, func)
    self.inputs[index] = func
end

function CGGreaterThan:getOutput(index)
    return self.inputs[0]() > self.inputs[1]()
end

return CGGreaterThan

File: Graph/ScriptTemplate/GreaterThan/CGGreaterThan.js
/**
 * @file CGGreaterThan.js
 * @author
 * @date 2021/8/15
 * @brief CGGreaterThan.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGGreaterThan extends BaseNode {
  constructor() {
    super();
  }

  getOutput() {
    return this.inputs[0]() > this.inputs[1]();
  }
}

exports.CGGreaterThan = CGGreaterThan;
